source/brailleDisplayDrivers/albatross/threads.py

```diff
@@ -0,0 +1,165 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2022 NV Access Limited, Burman's Computer and Education Ltd.
+
+"""Threads for Tivomatic Caiku Albatross braille display driver.
+Classes:
+- L{ReadThread}
+- L{RepeatedTimer}
+"""
+
+import ctypes
+import serial
+
+from ctypes import byref
+from ctypes.wintypes import DWORD
+from logHandler import log
+from serial.win32 import (
+	ERROR_IO_PENDING,
+	EV_RXCHAR,
+	GetLastError,
+	SetCommMask
+)
+from threading import (
+	Event,
+	Thread,
+	Timer
+)
+from typing import Callable
+
+import api
+
+from IAccessibleHandler import SecureDesktopNVDAObject
+
+from .constants import KC_INTERVAL
+
+
+class ReadThread(Thread):
+	"""Controls most of read operations and tries to reconnect when needed."""
+
+	def __init__(
+			self,
+			readFunction: Callable,
+			disableFunction: Callable,
+			event: Event,
+			dev: serial.Serial,
+			*args,
+			**kwargs
+	):
+		"""Constructor.
+		@param readFunction: Handles read operations and reconnection.
+		@param disableFunction: Called on connection failure.
+		@param event: Exit thread when set.
+		@param dev: Port object.
+		"""
+		super().__init__(*args, **kwargs)
+		self._readFunction = readFunction
+		self._disableFunction = disableFunction
+		self._event = event
+		self._dev = dev
+
+	def run(self):
+		data = dwEvtMask = DWORD()
+		log.debug(f"{self.name} started")
+		while not self._event.isSet():
+			if isinstance(api.getFocusObject(), SecureDesktopNVDAObject):

```

Comment:
these references to `SecureDesktopNVDAObject` (i.e. incorrect handling of #2315) should be removed

Reply:
> these references to `SecureDesktopNVDAObject` (i.e. incorrect handling of #2315) should be removed

As driver specific solution it did what I wanted, displaying UAC window on braille.

Reply:
Just to be explicit, this PR will not be merged with how this handles #2315

Reply:
> Just to be explicit, this PR will not be merged with how this handles #2315

I had removed it in my last commit. Unfortunately I'm not so familiar with git. Please let me know if it is still there.